provisioner/template.go

```diff
@@ -686,13 +698,34 @@ func zoneDistributedNodePoolGroups(nodePools []*api.NodePool) map[string]bool {
 		}
 	}
 
-	result := make(map[string]bool)
+	return poolGroups
+}
+
+// nodeLifeCycleProviderPerNodePoolGroup groups node-pools by the dedicated label, each node-pool-group is mapped to
+// the provider suitable for its node-pool profile
+// in case all pools of a group do not share the same profile, the group is mapped to empty string
```

Comment:
What does this mean in practice? Which provider would be used here by admission-controller? :thinking:

Reply:
an empty string "" provider will be used, which is practically turning this feature off